Runtime/Native/Android/NativeClient.cs

```diff
@@ -301,42 +317,50 @@ public void HandleAnr(string gameObjectName, string callbackName)
             _anrThread = new Thread(() =>
             {
                 float lastUpdatedCache = 0;
-                while (true)
+                while (_anrThread.IsAlive && _stopAnr == false)
                 {
-                    if (lastUpdatedCache == 0)
+                    if (!_preventAnr)
                     {
-                        lastUpdatedCache = _lastUpdateTime;
-                    }
-                    else if (lastUpdatedCache == _lastUpdateTime)
-                    {
-                        if (!reported)
+                        if (lastUpdatedCache == 0)
                         {
-
-                            reported = true;
-                            if (AndroidJNI.AttachCurrentThread() == 0)
+                            lastUpdatedCache = _lastUpdateTime;
+                        }
+                        else if (lastUpdatedCache == _lastUpdateTime)
+                        {
+                            if (!reported)
                             {
-                                // set temporary attribute to "Hang"
-                                AddAttribute(
-                                    AndroidJNI.NewStringUTF("error.type"),
-                                    AndroidJNI.NewStringUTF("Hang"));
-
-                                NativeReport(AndroidJNI.NewStringUTF("ANRException: Blocked thread detected."));
-                                // update error.type attribute in case when crash happen 
-                                SetAttribute("error.type", "Crash");
+
+                                reported = true;
+                                if (AndroidJNI.AttachCurrentThread() == 0)
+                                {
+                                    // set temporary attribute to "Hang"
+                                    AddAttribute(
+                                        AndroidJNI.NewStringUTF("error.type"),
+                                        AndroidJNI.NewStringUTF("Hang"));
```

Comment:
[General Note.  Making a change not necessary for this pull request] Reading through this made me think:  Why are your attribute strings defined in line like this?  In your editor code, you have an entire file where you define all of the Labels in one place, and yet these are in line.  Seems like following a similar pattern would be a good change both from the perspective of organization AND maintaining documentation.  If you have all of the attribute definitions in one place, would be easier for us to ensure any docs are up to date + give users a place in the code to see all of the attributes + quickly link to how they are used.  Let's discuss this as an aside

Reply:
It should be defined in const variables in class definition I agree - but this change might be important/usefull once we will merge implementations for native clients. I think it't an item on my to do list while we will work on native client refactor.

Reply:
Ok sounds good.  Yeah I view this change as primarily a documentation improvement than anything else, so we should just be keeping this in the back of our minds + slot in the work when and where it makes sense.
@vlussenburg